enhancements/installer/azure-cluster-scaling-for-outbound-traffic.md

```diff
@@ -0,0 +1,214 @@
+through the terraform-created public IP addresses and egress will work.
+
+
+### API Extensions
+
+This enhancment would modify the [CloudProviderConfig API Object](https://github.com/openshift/installer/blob/master/pkg/asset/manifests/azure/cloudproviderconfig.go) to include the property `disableOutboundSNAT` in the config.
+
+```go
+
+config := config{
+		authConfig: authConfig{
+			Cloud:                       params.CloudName.Name(),
+			TenantID:                    params.TenantID,
+			SubscriptionID:              params.SubscriptionID,
+			UseManagedIdentityExtension: true,
+			UserAssignedIdentityID: "",
+		},
+		ResourceGroup:     params.ResourceGroupName,
+		Location:          params.GroupLocation,
+		SubnetName:        params.SubnetName,
+		SecurityGroupName: params.NetworkSecurityGroupName,
+		VnetName:          params.VirtualNetworkName,
+		VnetResourceGroup: params.NetworkResourceGroupName,
+		RouteTableName:    params.ResourcePrefix + "-node-routetable",
+		CloudProviderRateLimit:       false,
+		CloudProviderBackoff:         true,
+		CloudProviderBackoffDuration: 6,
+
+		UseInstanceMetadata: true,
+		LoadBalancerSku:             "standard",
+		ExcludeMasterFromStandardLB: &excludeMasterFromStandardLB,
+    DisableOutboundSNAT: true, // <-- Add this property
+	}
+
+
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+There is a risk that a customer with a public that the customer could set disableOutboundSNAT 
+to true, without having a public IP address to use. We propose having the installer create 
+IP addresses through terraform to mitgate this risk.
+
+### Drawbacks
+
+The complexity of this task is percieved to be quite small. However it does have the drawback
+of having yet another configuration option to explain to the customer. This property is
+exposed to the customer in Azure already, so a customer might come across it anyway.
+
+## Design Details
+
+### Open Questions [optional]
+
+
+### Test Plan
+
+The [tests for the CloudProviderConfig API object](https://github.com/openshift/installer/blob/master/pkg/asset/manifests/azure/cloudproviderconfig_test.go) should be updated to the new default behavior.
+
+### Graduation Criteria
+
+
+#### Dev Preview -> Tech Preview
+
+
+#### Tech Preview -> GA
+
+
+#### Removing a deprecated feature
+
+
+### Upgrade / Downgrade Strategy
+
+This is only exposing a property that already exists in the generated cloud provider config 
+to the customer. The upgrade / downgrade should be a non-issue. 
+
+### Version Skew Strategy
+
+Version skew should not be an issue for this enhancement. 
+
+### Operational Aspects of API Extensions
+
+Sets the property `disableOutboundSNAT` to true in the cloud provider config.
+
+#### Failure Modes
+
+
+#### Support Procedures
+
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Alternatives
```

Comment:
Is there any merit in including NAT gateways as an alternative since the installer team are already looking into this?